src/set.rs

```diff
@@ -276,6 +276,93 @@ where
     {
         self.map.get_key_value(value, guard).map(|(k, _)| k)
     }
+
+    /// Returns `true` if `self` has no elements in common with `other`.
+    /// This is equivalent to checking for an empty intersection.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::iter::FromIterator;
+    /// use flurry::HashSet;
+    ///
+    /// let a = HashSet::from_iter(&[1, 2, 3]);
+    /// let b = HashSet::new();
+    /// let guard = a.guard();
+    ///
+    /// assert_eq!(a.is_disjoint(&b, &guard), true);
+    /// b.insert(4, &guard);
+    /// assert_eq!(a.is_disjoint(&b, &guard), true);
+    /// b.insert(1, &guard);
+    /// assert_eq!(a.is_disjoint(&b, &guard), false);
+    ///
+    /// ```
+    pub fn is_disjoint(&self, other: &HashSet<T, S>, guard: &Guard) -> bool {
+        for value in self.iter(guard) {
+            if other.contains(&value, guard) {
```

Comment:
Hmm, this probably wont' work down the line when we have non-global collectors, since a guard for one map/set won't work on a different one. Probably not worth fixing for now though.

Reply:
Oh, I see. Is there a workaround? Maybe we should use two different guards?

Reply:
If that's the case, seems like quite an easy fix, I think it's worth it.